Incorrect line number for Z331

# Bug report

## What's wrong

```python
def foreign_jwt_authentication(token: bytes):  # noqa: Z331
    """
    Goes to the remote provider to validate the token.

    Returns the payload if everything is fine.
    Otherwise raises an exception.
    """
    unverified_payload = jwt.decode(  # type: ignore
        token, key=None, verify=False,
    )
    _make_authentication_request(settings.TOKEN_VERIFICATION_URL, token)
    return unverified_payload
```

Error:

```
./server/rest_app/logics/auth.py:41:1: Z331 Found local variable that are only used in `return` statements
```

## How is that should be

Should be reported on the ` return unverified_payload` line.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -15,6 +15,10 @@ We used to have incremental versioning before `0.1.0`.
 - Adds `flake8-rst-docstrings` as a dependency
 - Limits yield tuple lengths
 
+### Bugfixes
+
+- Fixes incorrect line number for `Z331`
+
 
 ## 0.9.0
 

--- a/tests/fixtures/noqa.py
+++ b/tests/fixtures/noqa.py
@@ -262,9 +262,9 @@ try:
 except Exception as ex:  # noqa: Z329
     raise ex
 
-def some_function():  # noqa: Z331
+def some_function():
     some_value = 1
-    return some_value
+    return some_value  # noqa: Z331
 
 try:
     anti_z444 = 1

--- a/wemake_python_styleguide/visitors/ast/keywords.py
+++ b/wemake_python_styleguide/visitors/ast/keywords.py
@@ -2,7 +2,7 @@
 
 import ast
 from collections import defaultdict
-from typing import ClassVar, Dict, List, Type, Union
+from typing import ClassVar, Dict, List, Tuple, Type, Union
 
 from typing_extensions import final
 
@@ -26,6 +26,10 @@ from wemake_python_styleguide.visitors.base import BaseNodeVisitor
 from wemake_python_styleguide.visitors.decorators import alias
 
 AnyWith = Union[ast.With, ast.AsyncWith]
+NamesAndReturns = Tuple[
+    Dict[str, List[ast.Name]],
+    Dict[str, ast.Return],
+]
 ReturningViolations = Union[
     Type[InconsistentReturnViolation],
     Type[InconsistentYieldViolation],
@@ -252,26 +256,16 @@ class ConsistentReturningVariableVisitor(BaseNodeVisitor):
     def _get_return_node_variables(
         self,
         node: List[ast.AST],
-    ) -> Dict[str, List[ast.Name]]:
+    ) -> NamesAndReturns:
         returns: Dict[str, List[ast.Name]] = defaultdict(list)
-        for sub_node in node:
-            if isinstance(sub_node, ast.Return):
-                if isinstance(sub_node.value, ast.Name):
-                    variable_name = sub_node.value.id
-                    returns[variable_name].append(sub_node.value)
-        return returns
-
-    def _get_return_nodes(
-        self,
-        node: List[ast.AST],
-    ) -> Dict[str, ast.Return]:
         return_sub_nodes: Dict[str, ast.Return] = defaultdict()
         for sub_node in node:
             if isinstance(sub_node, ast.Return):
                 if isinstance(sub_node.value, ast.Name):
                     variable_name = sub_node.value.id
+                    returns[variable_name].append(sub_node.value)
                     return_sub_nodes[variable_name] = sub_node
-        return return_sub_nodes
+        return returns, return_sub_nodes
 
     def _check_variables_for_return(self, node: AnyFunctionDef) -> None:
         nodes = list(
@@ -282,8 +276,7 @@ class ConsistentReturningVariableVisitor(BaseNodeVisitor):
         )
         assign = self._get_assign_node_variables(nodes)
         names = self._get_name_nodes_variable(nodes)
-        returns = self._get_return_node_variables(nodes)
-        return_sub_nodes = self._get_return_nodes(nodes)
+        returns, return_sub_nodes = self._get_return_node_variables(nodes)
 
         returns = {name: returns[name] for name in returns if name in assign}
 

--- a/wemake_python_styleguide/visitors/ast/keywords.py
+++ b/wemake_python_styleguide/visitors/ast/keywords.py
@@ -261,6 +261,18 @@ class ConsistentReturningVariableVisitor(BaseNodeVisitor):
                     returns[variable_name].append(sub_node.value)
         return returns
 
+    def _get_return_nodes(
+        self,
+        node: List[ast.AST],
+    ) -> Dict[str, ast.Return]:
+        return_sub_nodes: Dict[str, ast.Return] = defaultdict()
+        for sub_node in node:
+            if isinstance(sub_node, ast.Return):
+                if isinstance(sub_node.value, ast.Name):
+                    variable_name = sub_node.value.id
+                    return_sub_nodes[variable_name] = sub_node
+        return return_sub_nodes
+
     def _check_variables_for_return(self, node: AnyFunctionDef) -> None:
         nodes = list(
             filter(
@@ -271,12 +283,20 @@ class ConsistentReturningVariableVisitor(BaseNodeVisitor):
         assign = self._get_assign_node_variables(nodes)
         names = self._get_name_nodes_variable(nodes)
         returns = self._get_return_node_variables(nodes)
+        return_sub_nodes = self._get_return_nodes(nodes)
 
         returns = {name: returns[name] for name in returns if name in assign}
 
+        self._check_for_violations(names, return_sub_nodes, returns)
+
+    def _check_for_violations(self, names, return_sub_nodes, returns) -> None:
         for variable_name in returns:
             if not set(names[variable_name]) - set(returns[variable_name]):
-                self.add_violation(InconsistentReturnVariableViolation(node))
+                self.add_violation(
+                    InconsistentReturnVariableViolation(
+                        return_sub_nodes[variable_name],
+                    ),
+                )
 
     def visit_return_variable(self, node: AnyFunctionDef) -> None:
         """

